ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py

```diff
@@ -133,6 +133,34 @@ def ethernet_interfaces(self) -> list | None:
                             "link_tracking_groups": link.get("link_tracking_groups"),
                         }
                     )
+            elif link["type"] == "underlay_p2p_vrfs":
+                ethernet_interface.update(
+                    {
+                        "mtu": self.shared_utils.p2p_uplinks_mtu,
+                        "type": "routed",
+                    }
+                )
+                for subinterface in get(link, "subinterfaces", default=[]):
+                    ethernet_subinterface = {
+                        "name": subinterface["interface"],
+                        "peer": link["peer"],
+                        "peer_interface": subinterface["peer_interface"],
+                        "peer_type": link["peer_type"],
+                        "vrf": vrf if (vrf := subinterface["vrf"]) != "default" else None,
+                        # TODO - make this bettter and part of the module
+                        "description": subinterface["vrf"],
+                        "shutdown": self.shared_utils.shutdown_interfaces_towards_undeployed_peers and not link["peer_is_deployed"],
+                        "type": "routed",
```

Comment:
We have a special type for l3 subinterface. If you don't set it, you will miss the encapsulation command.